### PR TITLE
Ft user can borrow books

### PR DESCRIPTION
**Pull request by AnayoOleru**
__What does this PR do?__
The ability for users to borrow a book

__Description of Task to be completed?__
create POST route to borrow a book
create controller that executes the routes
create a validation file to validate every data going into the bookHistory database 

__How should this be manually tested?__
Cloning this repo
npm install
create a .env file following the env sample
migrate your database
switch to branch ft-userCanBorrow-books
npm start
Test with postman

__Screenshots__
Screen Shot 2019-03-17 at 12 30 24 PM

### CODE REVIEW

File: server/src/controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import PostBorrowBook from './postborrowbook';
+
+
+const { borrowBook } = PostBorrowBook;
+
+export default borrowBook;
```
Comment (sojida) [inline server/src/controllers/index.js]: Please fix this End Of Line(EOL)

File: server/src/middlewares/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import BorrowBook from './postBorrowBook';
+
+ export default BorrowBook;
```
Comment (sojida) [inline server/src/middlewares/index.js]: EOL?

File: server/src/middlewares/postBorrowBook.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        error: 'Signup or signin to borrow this book',
+      });
+    }
+
+     req.data = user;
+    return next();
+}
+ }
+
+ export default BorrowBook;
```
Comment (sojida) [inline server/src/middlewares/postBorrowBook.js]: Great attention to detail here. Although you can fix this EOL.

File: server/src/middlewares/index.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+
```
Comment (sojida) [inline server/src/middlewares/index.js]: EOL here

File: server/src/middlewares/postBorrowBook.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
```
Comment (sojida) [inline server/src/middlewares/postBorrowBook.js]: EOL here

File: test/postBorrowBook.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+              returnon:"2019-04-09T11:00:00.000Z",
+          })
+          .end((error, res) => {
+            expect(res).to.be.json;
+            expect(res).to.have.status(400);
+            expect(res.body.error).to.equal("Id provided is invalid");
+            done();
+          });
+      });
+  });
```
Comment (sojida) [inline test/postBorrowBook.test.js]: EOL here

File: server/src/models/bookhistory.js [hunk 7]
```diff
@@ -26,17 +32,17 @@ module.exports = (sequelize, DataTypes) => {
...
-      allowNull: false,
+      allowNull: sequelize.NOW,
     },
   });
   bookhistory.associate = (models) => {
     // associations can be defined here
-    bookhistory.hasMany(models.books);
-    bookhistory.hasMany(models.users);
+    // bookhistory.hasMany(models.books);
+    // bookhistory.hasMany(models.users);
```
Comment (sojida) [inline server/src/models/bookhistory.js]: commented lines here

File: server/src/models/index.js [hunk 8]
```diff
@@ -32,4 +32,5 @@ db.sequelize = sequelize;
 db.Sequelize = Sequelize;
 
 
-export { db, config };
+// export { db, config };
```
Comment (sojida) [inline server/src/models/index.js]: commented line here

File: server/src/middlewares/index.js (hunk 4 above)
Comment (AnayoOleru) [reply server/src/middlewares/index.js]: There is already an EOL there

File: server/src/middlewares/postBorrowBook.js (hunk 5 above)
Comment (AnayoOleru) [reply server/src/middlewares/postBorrowBook.js]: Same with this

**Pull request merged**